### ISSUE TITLE
Live Editor: Look at possibility of adding background draft saving

### ISSUE THREAD

**Issue by Misplon** [siteorigin/siteorigin-panels#493]
Look at the possibility of adding background draft saving to the live editor.

**Comment by Misplon** [comment]
Sounds great! I, unfortunately, can't find the history for this issue on Slack. My bad, I should have provided a forum or Slack link. I think we're good to go. 

@AlexGStapleton If you have anything to add, please let us know. If not, please can you close.

**Comment by AlexGStapleton** [comment]
I can't find exact details but I remember the general reasoning behind this issue. A few premium users complained about losing work after working with the live editor and not saving the page. Basically, they would edit their pages using the live editor and work on them for a couple of hours. When they were done they would close the live editor and then close the page - thus not saving.

I think this issue was logged before the widget saving functionality was implemented so that in itself could be seen as a partial solution for existing widgets but it won't help in the case of new rows and widgets. A complete solution would be to manually trigger an autosave when closing the Live Editor.

**Comment by Misplon** [comment]
_manually trigger an autosave when closing the Live Editor_

Is that an option @gregpriday 

@AlexGStapleton When was the last time this came up?

**Comment by gregpriday** [comment]
That might be possible. I'll check what the mechanism is.

**Comment by Misplon** [comment]
Thanks!

**Comment by AlexGStapleton** [comment]
This comes up every two to three months. It's not very common but if it's a simple fix, we should consider doing this as this avoids a large amount of frustration for users who are affected and users thinking we have a buggy live editor. :(

Google suggests it's possible to autosave using `wp.autosave.server.triggerSave();`.

**Comment by gregpriday** [comment]
Just tested that out. It only seems to save the content of the main editor, not PB. You can actually save your work by switching to the main editor, then switching back to PB and have it copy your content.

Things also behave differently when it just saves and tries to restore from the browser cache. This doesn't work as well.

I'm sure there's a way to hack this, but we could just go with an interface change. Maybe in the Live Editor, instead of having a "Done" button, we have 2 buttons. "Close" and "Save". Close will close the Live Editor and Save will close it, then submit the entire post form.

Does that work for now? We can look at better autosave integration in the future.

**Comment by AlexGStapleton** [comment]
I like the idea of a dedicated saving button. I _think_ I might be mistaken and the issue was actually people thinking the Live Editor autosaved automatically and not only after saving the page (or clicking done).

**Comment by gregpriday** [comment]
Great, will go with that option.

**Comment by gregpriday** [comment]
Resolved in https://github.com/siteorigin/siteorigin-panels/commit/7a2c24b9fe24156b774ebabeeccaee0563aa50a5 - if we still have complaints about this issue, then we can open a new issue for adding background saving to the whole of Page Builder. This would apply to both the Live Editor and the main interface.
